src/Hyperion/ValueSerializers/ObjectSerializer.cs

```diff
@@ -103,7 +103,18 @@ public override void WriteManifest(Stream stream, SerializerSession session)
         public override void WriteValue(Stream stream, object value, SerializerSession session)
             => _writer(stream, value, session);
 
-        public override object ReadValue(Stream stream, DeserializerSession session) => _reader(stream, session);
+        public override object ReadValue(Stream stream, DeserializerSession session)
+        {
+            try
+            {
+                return _reader(stream, session);
+            }
+            catch (Exception e)
+            {
+                throw new Exception(
```

Comment:
Should this be a `SerializationException` or something more specific? Do we have any Hyperion-specific exception types?

Reply:
`SerializationException` doesn't exist in netstandard1.6 and net45

Reply:
I guess we can add `System.Runtime.Serialization.Primitives` package to add it in?